src/main/java/ch/njol/skript/expressions/ExprPlayerWeather.java

```diff
@@ -0,0 +1,95 @@
+/**
+ *  This file is part of Skript.
+ *
+ * Skript is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Skript is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * Copyright 2011-2017 Peter Güttinger and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.*;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.util.Getter;
+import ch.njol.skript.util.WeatherType;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.entity.Player;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * @author Nicofisi
+ * @author Peter Güttinger
+ */
+@Name("Player Weather")
+@Description("The weather for a player.")
+@Examples({"set weather for arg-player to clear",
+		"arg-player's weather is rainy"})
+@Since("2.2-dev34")
+public class ExprPlayerWeather extends PropertyExpression<Player, WeatherType> {
+	static {
+		Skript.registerExpression(ExprPlayerWeather.class, WeatherType.class, ExpressionType.PROPERTY,
+				"[the] weather [(for|of) %player%]", "%player%'[s] weather");
```

Comment:
this could cause issues with the normal weather expression, it should be registered using the normal registerPropertyExpression method

Reply:
Never used it before but sure, gimme a second

Reply:
Btw I actually copied that from ExprWeather, is it fine there?
```java
	static {
		Skript.registerExpression(ExprWeather.class, WeatherType.class, ExpressionType.PROPERTY, "[the] weather [(in|of) %worlds%]", "%worlds%'[s] weather");
	}
```